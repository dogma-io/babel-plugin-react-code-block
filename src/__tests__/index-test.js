import {transform} from '@babel/core'

function t(code, options) {
  options = options || {component: 'CodeBlock'}

  return transform(code, {
    plugins: [[require('../../lib/'), options]],
    presets: ['@babel/env', '@babel/react'],
  })
}

describe('babel-plugin-react-code-block', () => {
  it('should throw if component option is missing', () => {
    expect(() => {
      t('<CodeBlock />', {})
    }).toThrow(
      /babel-plugin-react-code-block expects component option to be present and to be a string/,
    )
  })

  it('should throw if component option is not a string', () => {
    expect(() => {
      t('<CodeBlock />', {component: 1})
    }).toThrow(
      /babel-plugin-react-code-block expects component option to be present and to be a string/,
    )
  })

  it('should inject code for code block component with children', () => {
    expect(
      t(
        'var a = <CodeBlock><div className="foobar" onClick={() => {}} /></CodeBlock>',
      ),
    ).toMatchSnapshot()
  })

  it('should inject code for code block component with children and use maxLineLength attribute with prettier', () => {
    expect(
      t(
        'var a = <CodeBlock maxLineLength={10}><div className="foobar" onClick={() => {}} /></CodeBlock>',
      ),
    ).toMatchSnapshot()
  })

  it('should not inject code for code block component without children', () => {
    expect(t('var a = <CodeBlock code="foobar" />')).toMatchSnapshot()
  })

  it('should not touch non-code block component component', () => {
    expect(
      t('var a = <NotCodeBlock><div><span /></div></NotCodeBlock>'),
    ).toMatchSnapshot()
  })
})
