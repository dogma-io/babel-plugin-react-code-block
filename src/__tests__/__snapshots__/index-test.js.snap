// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-react-code-block should inject code for code block component with children 1`] = `
"\\"use strict\\";

var a = React.createElement(CodeBlock, {
  code: \\"<div className=\\\\\\"foobar\\\\\\" onClick={() => {}} />\\"
}, React.createElement(\\"div\\", {
  className: \\"foobar\\",
  onClick: function onClick() {}
}));"
`;

exports[`babel-plugin-react-code-block should inject code for code block component with children and use maxLineLength attribute with prettier 1`] = `
"\\"use strict\\";

var a = React.createElement(CodeBlock, {
  maxLineLength: 10,
  code: \\"<div\\\\\\\\n  className=\\\\\\"foobar\\\\\\"\\\\\\\\n  onClick={() => {}}\\\\\\\\n/>\\"
}, React.createElement(\\"div\\", {
  className: \\"foobar\\",
  onClick: function onClick() {}
}));"
`;

exports[`babel-plugin-react-code-block should not inject code for code block component without children 1`] = `
"\\"use strict\\";

var a = React.createElement(CodeBlock, {
  code: \\"foobar\\"
});"
`;

exports[`babel-plugin-react-code-block should not touch non-code block component component 1`] = `
"\\"use strict\\";

var a = React.createElement(NotCodeBlock, null, React.createElement(\\"div\\", null, React.createElement(\\"span\\", null)));"
`;
